## Business Objective
#### To predict the salary of the candidate

In [1]:
# import library

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


## Dataset Details

In [2]:
# hiring.csv is downlaoded from kaggle
# import hiring.csv

hiring_df = pd.read_csv(r"D:\Project\CodeBasicsTasks\MultiLinearRegression_Hiring\hiring.csv")

In [3]:
hiring_df.shape


(8, 4)

In [5]:
list(hiring_df.columns)

['experience',
 'test_score(out of 10)',
 'interview_score(out of 10)',
 'salary($)']

In [6]:
hiring_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8 entries, 0 to 7
Data columns (total 4 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   experience                  6 non-null      object 
 1   test_score(out of 10)       7 non-null      float64
 2   interview_score(out of 10)  8 non-null      int64  
 3   salary($)                   8 non-null      int64  
dtypes: float64(1), int64(2), object(1)
memory usage: 384.0+ bytes


In [7]:
hiring_df.isna().sum()

experience                    2
test_score(out of 10)         1
interview_score(out of 10)    0
salary($)                     0
dtype: int64

In [10]:
hiring_df

experience  test_score(out of 10)  interview_score(out of 10)  salary($)
0       Zero                    8.0                           9      50000
1       Zero                    8.0                           6      45000
2       five                    6.0                           7      60000
3        two                   10.0                          10      65000
4      seven                    9.0                           6      70000
5      three                    7.0                          10      62000
6        ten                    NaN                           7      72000
7     eleven                    7.0                           8      80000

## Explaratory Data Anaysis 

#### Misisng Value Imputation

In [9]:
hiring_df["experience"] = hiring_df["experience"].fillna("Zero")

In [11]:
import math

In [17]:
mean_test_score = hiring_df["test_score(out of 10)"].mean()
mean_test_score =math.floor( mean_test_score)
mean_test_score

7

In [18]:
hiring_df["test_score(out of 10)"] = hiring_df["test_score(out of 10)"].fillna(mean_test_score)

In [19]:
hiring_df

experience  test_score(out of 10)  interview_score(out of 10)  salary($)
0       Zero                    8.0                           9      50000
1       Zero                    8.0                           6      45000
2       five                    6.0                           7      60000
3        two                   10.0                          10      65000
4      seven                    9.0                           6      70000
5      three                    7.0                          10      62000
6        ten                    7.0                           7      72000
7     eleven                    7.0                           8      80000

In [23]:
#### Converting exeperince value to numbers
from word2number import w2n

In [27]:
hiring_df.experience = hiring_df.experience.apply(w2n.word_to_num)

In [28]:
hiring_df

experience  test_score(out of 10)  interview_score(out of 10)  salary($)
0           0                    8.0                           9      50000
1           0                    8.0                           6      45000
2           5                    6.0                           7      60000
3           2                   10.0                          10      65000
4           7                    9.0                           6      70000
5           3                    7.0                          10      62000
6          10                    7.0                           7      72000
7          11                    7.0                           8      80000

In [30]:
hiring_df.describe()

experience  test_score(out of 10)  interview_score(out of 10)  \
count     8.00000                8.00000                    8.000000   
mean      4.75000                7.75000                    7.875000   
std       4.26782                1.28174                    1.642081   
min       0.00000                6.00000                    6.000000   
25%       1.50000                7.00000                    6.750000   
50%       4.00000                7.50000                    7.500000   
75%       7.75000                8.25000                    9.250000   
max      11.00000               10.00000                   10.000000   

         salary($)  
count      8.00000  
mean   63000.00000  
std    11501.55269  
min    45000.00000  
25%    57500.00000  
50%    63500.00000  
75%    70500.00000  
max    80000.00000

## Model Building

In [29]:
from sklearn.linear_model import LinearRegression

In [33]:
hiring_model = LinearRegression()
hiring_model.fit(hiring_df[["experience","test_score(out of 10)","interview_score(out of 10)"]],hiring_df["salary($)"])

LinearRegression()

In [35]:
hiring_model.coef_

array([2922.26901502, 2221.30909959, 2147.48256637])

In [40]:
hiring_model.intercept_

14992.65144669314

#### Model Equation
#### salary = 14992.65 + 2922.27 * experience + 2221.30909959 * test_score + 2147.48 * interview_score

In [41]:
from sklearn.metrics import mean_squared_error

In [43]:
predicted_salary = hiring_model.predict(hiring_df[["experience","test_score(out of 10)","interview_score(out of 10)"]])

In [45]:
mse = mean_squared_error(hiring_df["salary($)"],predicted_salary)
mse

2660640.9097884092

In [47]:
rmse = math.sqrt(mse)
rmse

1631.1471146982449

In [ ]:
## Predict for some unknown values


#### experience= 2
#### test_score(out of 10) = 9 
#### interview_score(out of 10) =6 

In [53]:
salary  = hiring_model.predict([[2,9,6]])
salary[0]

53713.86677124005

#### experience= 12
#### test_score(out of 10) = 10 
#### interview_score(out of 10) = 10

In [54]:
salary  = hiring_model.predict([[12,10,10]])
salary[0]

93747.79628650834